Developer-Guide--Writing-Tests.md

```diff
@@ -311,3 +317,142 @@ end
+
+On a high level, RTL allows you to test from the user's perspective, while Enzyme gives you access to the internal states/implementation of the components.
+
+Similar to RSpec, Jest also involves specifications. It is recommended that you follow along the `__tests__` folder, under `app/assets/javascripts/Components/`, for this tutorial.
+
+## How to Run Jest Specifications
+
+**Note:** The following commands assume you are within the Markus root folder.
+
+`yarn` is the package manager we use in Markus. Its counterpart `npm` is sometimes more known.
+
+To run all specifications:
+
+```sh
+yarn test
+```
+
+To run all specifications with the test-coverage table shown:
+
+```sh
+yarn test-cov
+```
+
+To run a specific specification:
+
+```sh
+yarn test <filename>
+```
+
+For example, to run the `StudentTable` specification, run `yarn test student_table.test.jsx`.
+
+These commands are specified in `package.json`, under Markus root.
+
+## Jest Configurations
+
+There are a lot of [configurations](https://jestjs.io/docs/configuration) within Jest. The main source of configuration is jest.config.js under the Markus root. Most of the settings are left as default, and each setting has a comment explaining what they are for. Make sure you update the respective settings when needed. We'll cover 3 important ones below.
+
+### setupFiles
+
+This setting points to a list of files that are run to set up the testing environment. For instance, the imports of `JQuery` and `I18n` would be in it.
+
+### setupFilesAfterEnv
+
+This setting points to a list of files that are run immediately after the setup of testing environment, before the actual tests. This is a great place for your global imports in the test files, such as the import of `React`, the configuration of `Enzyme`.
+
+### testMatch
+
+This setting points to a list of patterns that entails directories/modules you want Jest to look at to find your tests. For instance, `**/__tests__/**/*.[jt]s?(x)` would include the path `app/assets/javascripts/Components/__tests__/student_table.test.jsx`.
+
+## Jest Naming Conventions
+
+### Folders
+
+Currently we use a centralized `__tests__` folder under `components`. The folder is intended to include tests for all the React components.
+
+### Files
+
+The name of the file should have the format `<component>.test.jsx`. For instance, the test file for the StudentTable component would be `student_table.test.jsx`. Lower cases are recommended so the `_` symbol should be used to separate words.
```

Comment:
You can remove the sentence about "Lower cases...", since this is a convention we use across the whole code base (not specific to test files).